### ISSUE TITLE
Time out in waiting for serviced.log (EC2 Ubuntu server instance)

### ISSUE THREAD

**Issue by schniedergers** [monitoringartist/zenoss5-core-autodeploy#11]
I created a brand new Ubuntu 14.04 EC2 instance, added 4 partitions, and ran the core-autodeploy.sh as: "./core-autodeploy.sh -d /dev/xvdb1 -s /dev/xvdb2 -v /dev/xvdb3 -b /dev/xvdb4"

Result is at http://pastebin.com/dX3H6yhg.

It never gets to finding anything in /var/log/upstart/serviced.log (file does not exist) and times out.

The only caveat is that I have only 2 CPUs instead of 4, 8GB RAM instead of 20GB, and some slightly smaller partitions. But that should not be a problem during installation, right?

**Comment by jangaraj** [comment]
It's not a issue for now:
```
#2: This is not a problem, because Control Centre service is not fully started, I'm trying in 10 seconds
Message from author of autodeploy script: Keep calm and be patient! - http://www.keepcalmandposters.com/posters/38112.png
```
You was waiting only 20seconds. If this loop will finish after 15 minutes with error, only then it's real issue. Information has been printed:
```
4.1 Adding current host to the default resource pool
Please be patient, because docker image zenoss/serviced-isvcs must be downloaded before first start.
You can check progress in new console: tailf /var/log/upstart/serviced.log
Script is trying to check status every 10s. Timeout for this step is 15 minutes.
```
Pls ignore "tail: cannot open ‘/var/log/upstart/serviced.log’ for reading: No such file or directory"

Please be patient and then please update this issue.

**Comment by schniedergers** [comment]
I waited the whole 15 minutes (until the script timed out itself) - just didn't paste the complete log as it was just repeating the same message.

The last lines were:
```
Message from author of autodeploy script: Keep calm and be patient! - http://www.keepcalmandposters.com/posters/38112.png
Last serviced log line: tail: cannot open ‘/var/log/upstart/serviced.log’ for reading: No such file or directory#89: This is not a problem, because Control Centre service is not fully started, I'm trying in 10 seconds
Message from author of autodeploy script: Keep calm and be patient! - http://www.keepcalmandposters.com/posters/38112.png
Last serviced log line: tail: cannot open ‘/var/log/upstart/serviced.log’ for reading: No such file or directory
echo "could not create a client to the master: dial tcp 172.29.0.21:4979: connection refused" | grep -v loopback | wc -l
Problem with adding a host - check output from test: could not create a client to the master: dial tcp 172.29.0.21:4979: connection refused
```

**Comment by jangaraj** [comment]
Perfect. Log file should be somewhere, try to find where (e.g. find / -name "serviced.log") please. Hopefully, something will be there.

**Comment by rhyven** [comment]
Same here on a fresh install of Ubuntu 14.04.02 LTS.  Also no serviced.log file anywhere  :(

    #89: This is not a problem, because Control Centre service is not fully started, I'm trying in 10 seconds
    Message from author of autodeploy script: Keep calm and be patient! - http://www.keepcalmandposters.com/posters/38112.png
    Last serviced log line: tail: cannot open ‘/var/log/upstart/serviced.log’ for reading: No such file or directory
    echo "could not create a client to the master: dial tcp 10.7.140.3:4979: connection refused" | grep -v loopback | wc -l
    Problem with adding a host - check output from test: could not create a client to the master: dial tcp 10.7.140.3:4979: connection refused
    root@***:/var/log# find / -name "serviced.log"
    root@***:/var/log#


**HOWEVER**, there is an empty folder at /var/log/serviced:

    root@***:/var/log# ls -la | grep serviced
    drwxrwxrwt  2 root      root     4096 Jul 16 17:17 serviced

Does this help you track down the problem?

**Comment by rhyven** [comment]
Further info:

Ran `sudo apt-get update && sudo apt-get upgrade`, then rebooted.  Still getting errors, but slightly different:

    could not create a client to the master: dial tcp 10.7.140.3:4979: connection refused
    #41: This is not a problem, because Control Centre service is not fully started, I'm trying in 10 seconds
    Message from author of autodeploy script: Keep calm and be patient! - http://www.keepcalmandposters.com/posters/38112.png
    Last serviced log line: Thu Jul 16 17:49:03 NZST 2015: serviced is now stopped - done with post-stop

/var/log/upstart/serviced.log contains:

    Thu Jul 16 17:47:32 NZST 2015: starting serviced daemon for 070a038c - waiting for docker
    CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
    Thu Jul 16 17:47:33 NZST 2015: docker is now ready - done with pre-start
    docker0   Link encap:Ethernet  HWaddr 56:84:7a:fe:97:99  
              inet addr:172.17.42.1  Bcast:0.0.0.0  Mask:255.255.0.0
              UP BROADCAST MULTICAST  MTU:1500  Metric:1
              RX packets:0 errors:0 dropped:0 overruns:0 frame:0
              TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:0 
              RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

    eth0      Link encap:Ethernet  HWaddr 82:40:6c:7c:23:3a  
              inet addr:10.7.140.3  Bcast:10.7.140.255  Mask:255.255.255.0
              inet6 addr: fe80::8040:6cff:fe7c:233a/64 Scope:Link
              UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
              RX packets:213 errors:0 dropped:39 overruns:0 frame:0
              TX packets:46 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:1000 
              RX bytes:21194 (21.1 KB)  TX bytes:5058 (5.0 KB)
    
    lo        Link encap:Local Loopback  
              inet addr:127.0.0.1  Mask:255.0.0.0
              inet6 addr: ::1/128 Scope:Host
              UP LOOPBACK RUNNING  MTU:65536  Metric:1
              RX packets:22 errors:0 dropped:0 overruns:0 frame:0
              TX packets:22 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:0 
              RX bytes:2872 (2.8 KB)  TX bytes:2872 (2.8 KB)
    
    + . /etc/default/serviced
    + HOME=/root
    + SERVICED_AGENT=1
    + SERVICED_MASTER=1
    + SERVICED_REGISTRY=1
    + SERVICED_FS_TYPE=btrfs
    + set +x
    exporting HOME
    exporting SERVICED_AGENT
    exporting SERVICED_MASTER
    exporting SERVICED_REGISTRY
    exporting SERVICED_FS_TYPE
    This master has been configured to be in pool: default
    I0716 05:47:34.673593 01350 api.go:115] StartServer: [] (0)
    I0716 05:47:34.724014 01350 daemon.go:175] Listening on [::]:4979
    I0716 05:47:34.724952 01350 daemon.go:221] Checking interface address at 10.7.140.3/24
    I0716 05:47:34.725277 01350 daemon.go:224] Host FFF-MONITOR is a local address at 10.7.140.3
    I0716 05:47:34.728160 01350 manager.go:123] setting elasticsearch-serviced, cluster: 3aajkn1rh21v0hkc7jjv46cr7
    I0716 05:47:34.728511 01350 manager.go:123] setting elasticsearch-logstash, cluster: 3vgacw4s03sc3oo6ae0kdxs7
    I0716 05:47:34.730013 01350 manager.go:233] Looking for image zenoss/serviced-isvcs:v27.1 in tar /opt/serviced/images/zenoss/serviced-isvcs/v27.1.tar.gz
    I0716 05:47:34.730133 01350 manager.go:244] Pulling image zenoss/serviced-isvcs:v27.1
    E0716 05:49:03.889076 01350 api.go:690] Unable to pull image zenoss/serviced-isvcs:v27.1
    F0716 05:49:03.889267 01350 daemon.go:145] Could not start isvcs: Failed to pull image zenoss/serviced-isvcs:v27.1: image zenoss/serviced-isvcs:v27.1 not available
    Thu Jul 16 17:49:03 NZST 2015: post-stopping serviced daemon - waiting for serviced to stop
    Thu Jul 16 17:49:03 NZST 2015: waiting for serviced daemon to stop
    Thu Jul 16 17:49:03 NZST 2015: waiting for serviced to stop listening
    Thu Jul 16 17:49:03 NZST 2015: waiting for serviced isvcs to stop
    time="2015-07-16T17:49:03+12:00" level="fatal" msg="Cannot connect to the Docker daemon. Is 'docker -d' running on this host?" 
    Thu Jul 16 17:49:03 NZST 2015: serviced is now stopped - done with post-stop

**Comment by rhyven** [comment]
it@***:/var/log$ sudo netstat -npl
    Active Internet connections (only servers)
    Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
    tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      792/rpcbind     
    tcp        0      0 0.0.0.0:58096           0.0.0.0:*               LISTEN      4661/rpc.mountd 
    tcp        0      0 127.0.0.1:9200          0.0.0.0:*               LISTEN      3762/docker-proxy
    tcp        0      0 0.0.0.0:52113           0.0.0.0:*               LISTEN      4661/rpc.mountd 
    tcp        0      0 0.0.0.0:44754           0.0.0.0:*               LISTEN      -               
    tcp        0      0 127.0.0.1:12181         0.0.0.0:*               LISTEN      3729/docker-proxy
    tcp        0      0 0.0.0.0:53              0.0.0.0:*               LISTEN      1097/dnsmasq    
    tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1045/sshd       
    tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      3698/docker-proxy
    tcp        0      0 0.0.0.0:47966           0.0.0.0:*               LISTEN      817/rpc.statd   
    tcp        0      0 0.0.0.0:46560           0.0.0.0:*               LISTEN      4661/rpc.mountd 
    tcp        0      0 0.0.0.0:2049            0.0.0.0:*               LISTEN      -               
    tcp        0      0 127.0.0.1:9090          0.0.0.0:*               LISTEN      3810/docker-proxy
    tcp        0      0 127.0.0.1:58888         0.0.0.0:*               LISTEN      3639/docker-proxy
    tcp        0      0 127.0.0.1:58443         0.0.0.0:*               LISTEN      3887/docker-proxy
    tcp        0      0 127.0.0.1:9292          0.0.0.0:*               LISTEN      5060/docker-proxy
    tcp        0      0 127.0.0.1:9100          0.0.0.0:*               LISTEN      3681/docker-proxy
    tcp6       0      0 :::111                  :::*                    LISTEN      792/rpcbind     
    tcp6       0      0 :::50000                :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::80                   :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::5042                 :::*                    LISTEN      5068/docker-proxy
    tcp6       0      0 :::4242                 :::*                    LISTEN      3866/docker-proxy
    tcp6       0      0 :::5043                 :::*                    LISTEN      5077/docker-proxy
    tcp6       0      0 :::4979                 :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::43125                :::*                    LISTEN      4661/rpc.mountd 
    tcp6       0      0 :::53                   :::*                    LISTEN      1097/dnsmasq    
    tcp6       0      0 :::6006                 :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::22                   :::*                    LISTEN      1045/sshd       
    tcp6       0      0 :::443                  :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::8443                 :::*                    LISTEN      3654/docker-proxy
    tcp6       0      0 :::44225                :::*                    LISTEN      4661/rpc.mountd 
    tcp6       0      0 :::2049                 :::*                    LISTEN      -               
    tcp6       0      0 :::38625                :::*                    LISTEN      817/rpc.statd   
    tcp6       0      0 :::2181                 :::*                    LISTEN      3832/docker-proxy
    tcp6       0      0 :::7878                 :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::5000                 :::*                    LISTEN      3628/docker-proxy
    tcp6       0      0 :::22250                :::*                    LISTEN      3159/serviced   
    tcp6       0      0 :::35787                :::*                    LISTEN      4661/rpc.mountd 
    tcp6       0      0 :::60045                :::*                    LISTEN      -               
    udp        0      0 0.0.0.0:59956           0.0.0.0:*                           4661/rpc.mountd 
    udp        0      0 0.0.0.0:53              0.0.0.0:*                           1097/dnsmasq    
    udp        0      0 0.0.0.0:111             0.0.0.0:*                           792/rpcbind     
    udp        0      0 172.17.42.1:123         0.0.0.0:*                           1623/ntpd       
    udp        0      0 10.7.140.3:123          0.0.0.0:*                           1623/ntpd       
    udp        0      0 127.0.0.1:123           0.0.0.0:*                           1623/ntpd       
    udp        0      0 0.0.0.0:123             0.0.0.0:*                           1623/ntpd       
    udp        0      0 0.0.0.0:33602           0.0.0.0:*                           -               
    udp        0      0 0.0.0.0:898             0.0.0.0:*                           792/rpcbind     
    udp        0      0 127.0.0.1:996           0.0.0.0:*                           817/rpc.statd   
    udp        0      0 0.0.0.0:50556           0.0.0.0:*                           4661/rpc.mountd 
    udp        0      0 0.0.0.0:34460           0.0.0.0:*                           817/rpc.statd   
    udp        0      0 0.0.0.0:2049            0.0.0.0:*                           -               
    udp        0      0 0.0.0.0:51258           0.0.0.0:*                           4661/rpc.mountd 
    udp6       0      0 :::36807                :::*                                4661/rpc.mountd 
    udp6       0      0 :::56312                :::*                                4661/rpc.mountd 
    udp6       0      0 :::45952                :::*                                -               
    udp6       0      0 :::48075                :::*                                817/rpc.statd   
    udp6       0      0 :::53                   :::*                                1097/dnsmasq    
    udp6       0      0 :::111                  :::*                                792/rpcbind     
    udp6       0      0 fe80::8d:dfff:fe3c::123 :::*                                1623/ntpd       
    udp6       0      0 fe80::c428:e4ff:fe6:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::1433:e0ff:fe1:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::c491:f4ff:fe7:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::3408:e6ff:fef:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::f886:5ff:fe14:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::b02e:f5ff:feb:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::5484:7aff:fef:123 :::*                                1623/ntpd       
    udp6       0      0 fe80::8040:6cff:fe7:123 :::*                                1623/ntpd       
    udp6       0      0 ::1:123                 :::*                                1623/ntpd       
    udp6       0      0 :::123                  :::*                                1623/ntpd       
    udp6       0      0 :::33388                :::*                                4661/rpc.mountd 
    udp6       0      0 :::898                  :::*                                792/rpcbind     
    udp6       0      0 :::2049                 :::*                                -               
    Active UNIX domain sockets (only servers)
    Proto RefCnt Flags       Type       State         I-Node   PID/Program name    Path
    unix  2      [ ACC ]     STREAM     LISTENING     13071    2008/docker         /var/run/docker.sock
    unix  2      [ ACC ]     STREAM     LISTENING     9544     1/init              @/com/ubuntu/upstart
    unix  2      [ ACC ]     STREAM     LISTENING     9077     1092/acpid          /var/run/acpid.socket
    unix  2      [ ACC ]     STREAM     LISTENING     10682    578/dbus-daemon     /var/run/dbus/system_bus_socket
    unix  2      [ ACC ]     SEQPACKET  LISTENING     9676     434/systemd-udevd   /run/udev/control
    unix  2      [ ACC ]     STREAM     LISTENING     8948     792/rpcbind         /run/rpcbind.sock

**Comment by rhyven** [comment]
Getting a little progress.

I noticed port 4979 was listening on ipv6, so I disabled ipv6 by adding the following lines to /etc/sysctl.conf:

    net.ipv6.conf.all.disable_ipv6 = 1
    net.ipv6.conf.default.disable_ipv6 = 1
    net.ipv6.conf.lo.disable_ipv6 = 1

Now I get the following: 

    4.1 Adding current host to the default resource pool
    Please be patient, because docker image zenoss/serviced-isvcs must be downloaded before first start.
    You can check progress in new console: tailf /var/log/upstart/serviced.log
    Script is trying to check status every 10s. Timeout for this step is 15 minutes.
    serviced host list 2>&1
    echo "error executing query Post http://localhost:9200/controlplane/host/_search?size=10000&version=true: dial tcp 127.0.0.1:9200: connection refused" | grep -v loopback | wc -l
    Problem with adding a host - check output from test: error executing query Post http://localhost:9200/controlplane/host/_search?size=10000&version=true: dial tcp 127.0.0.1:9200: connection refused

Running `sudo netstat -npl` confirms that port 9200 isn't listening anywhere.

However now I can access the Control Center by going to the IP address.

No more time tonight but will test more tomorrow  :)

**Comment by jangaraj** [comment]
I recommend manual installation for now.
Zenoss released new version of CC and Zenoss last week, so probably they
have changed something. At least rhel/centos installation was updated.

**Comment by rhyven** [comment]
Cool, will do.  Thanks @jangaraj 

Also, thanks for all your hard work  :)

**Comment by jangaraj** [comment]
Done. Ubuntu problem was that init script doesn't return correct exit code = serviced was always started for the script. Updates for 5.0.4 version has been also applied. Unfortunately install doc for Core 5.0.4 was not released, but I've used RM 5.0.4 doc. Feel free to test and close this issue. My today's Ubuntu/Redhat installations looks OK.
